deepmd/pt/model/task/dipole.py

```diff
@@ -1,67 +1,152 @@
 # SPDX-License-Identifier: LGPL-3.0-or-later
 import logging
+from typing import (
+    List,
+    Optional,
+)
 
 import torch
 
-from deepmd.pt.model.network.network import (
-    ResidualDeep,
+from deepmd.dpmodel import (
+    FittingOutputDef,
+    OutputVariableDef,
 )
 from deepmd.pt.model.task.fitting import (
-    Fitting,
+    GeneralFitting,
+)
+from deepmd.pt.utils import (
+    env,
+)
+from deepmd.pt.utils.env import (
+    DEFAULT_PRECISION,
 )
 
 log = logging.getLogger(__name__)
 
 
-class DipoleFittingNet(Fitting):
-    def __init__(
-        self, ntypes, embedding_width, neuron, out_dim, resnet_dt=True, **kwargs
-    ):
-        """Construct a fitting net for dipole.
+class DipoleFittingNet(GeneralFitting):
+    """Construct a general fitting net.
 
-        Args:
-        - ntypes: Element count.
-        - embedding_width: Embedding width per atom.
-        - neuron: Number of neurons in each hidden layers of the fitting net.
-        - bias_atom_e: Average enery per atom for each element.
-        - resnet_dt: Using time-step in the ResNet construction.
-        """
-        super().__init__()
-        self.ntypes = ntypes
-        self.embedding_width = embedding_width
-        self.out_dim = out_dim
+    Parameters
+    ----------
+    var_name : str
+        The atomic property to fit, 'dipole'.
+    ntypes : int
+        Element count.
+    dim_descrpt : int
+        Embedding width per atom.
+    dim_out : int
+        The output dimension of the fitting net.
+    dim_rot_mat : int
+        The dimension of rotation matrix, m1.
+    neuron : List[int]
+        Number of neurons in each hidden layers of the fitting net.
+    resnet_dt : bool
+        Using time-step in the ResNet construction.
+    numb_fparam : int
+        Number of frame parameters.
+    numb_aparam : int
+        Number of atomic parameters.
+    activation_function : str
+        Activation function.
+    precision : str
+        Numerical precision.
+    distinguish_types : bool
+        Neighbor list that distinguish different atomic types or not.
+    rcond : float, optional
+        The condition number for the regression of atomic energy.
+    seed : int, optional
+        Random seed.
+    """
 
-        filter_layers = []
-        one = ResidualDeep(
-            0, embedding_width, neuron, 0.0, out_dim=self.out_dim, resnet_dt=resnet_dt
+    def __init__(
+        self,
+        var_name: str,
+        ntypes: int,
+        dim_descrpt: int,
+        dim_rot_mat: int,
```

Comment:
please not my comment in #3286 , the `embedding_width` is preferred.

Reply:
if it's not fixed in #3286, I will fix it in the polar fitting pr.